### PR TITLE
feat(siwf): add siwf v2

### LINKED ISSUE #170: SIWF v2: User Login and Session Creation in SAT Backend
### Summary:
As a developer, I want to implement the logic to finalize user login and create a user session in the app backend, so the user can be authenticated after signing in with Frequency.

#### Acceptance Criteria:

- User clicks SIWF v2 button (Goes through the loop)
- Backend receives the callback from SIWF v2 and passes it on to Account Service.
- Provide appropriate feedback to the app frontend, such as redirecting to the post sign-in screen.
- Environment variable on SAT to switch between SIWF v1 and v2

#### Developer Notes

- For now, it MUST be OPTIONAL to use SIWF v2 instead of SIWF v1 as SIWF v2 doesn't work with a local network and requires approval before it is possible to use on Testnet

#### Tasks:

1. Implement backend logic to handle the callback from SIWF v2 server.
2. If the user is new, create a new user account in the app's database.
3. Establish a user session and manage session tokens or cookies as appropriate.
4. Send the necessary data back to the app frontend to update the UI.
5. Ensure proper error handling and logging.
6. Write unit and integration tests.

SIWF V2 Docs: https://projectlibertylabs.github.io/siwf/v2/docs/
SIWF v2 Gateway endpoints: https://projectlibertylabs.github.io/gateway/account/#tag/v2accounts
SIWF v2 with Account Service Example: https://github.com/ProjectLibertyLabs/gateway/blob/main/tools/siwfv2/example.mjs

### PR DESCRIPTION
**Pull request by enddynayn**
Introduces support for sign-in and sign-up using SIWF V2.

This update enables apps to onboard users with familiar login methods,
allowing users to authenticate with either their email or phone number,
bridging the experience for web2 users.

- Implemented backend routes to support the SIWF V2 flow
- Add a frontend button to navigate users to login with SIWF
- Created a callback page view to display account creation

siwf sat-2024-10-29-214721

Closes #170

### CODE REVIEW

File: backend/controllers/AuthController.ts [hunk 1]
```diff
@@ -19,9 +25,65 @@ export class AuthController extends BaseController {
...
+    }
+  }
+
+  public async getFrequencyAccessLogin(req: Request, res: Response) {
+    try {
+      const HOSTNAME = 'localhost';
+      const PORT = Config.instance().port;
+      const SIWF_CALLBACK = `http://${HOSTNAME}:${PORT}/login/callback`;
+
+      const params = {
```
Comment (enddynayn) [inline backend/controllers/AuthController.ts:62]: The client should provide this information, but we can refactor it in the future if necessary.

File: backend/controllers/AuthController.ts [hunk 2]
```diff
@@ -19,9 +25,65 @@ export class AuthController extends BaseController {
...
 
+  private validateQueryParams(params: AccountQueryParams): string | null {
+    const { msaId, referenceId, accountId } = params;
+    if (!msaId && !referenceId && !accountId) {
+      return 'msaId, referenceId, or accountId is required';
+    }
+    return null;
+  }
+
+  public async postLoginV2Swif(req: Request, res: Response) {
```
Comment (mattheworris) [inline backend/controllers/AuthController.ts:41]: typo:
```suggestion
  public async postLoginV2Siwf(req: Request, res: Response) {
```

File: backend/controllers/AuthController.ts [hunk 3]
```diff
@@ -19,9 +25,65 @@ export class AuthController extends BaseController {
...
+    } catch (e) {
+      if (e instanceof HttpError) {
+        res.status(e.code).send(e.message).end();
+      } else {
+        res.status(HttpStatusCode.InternalServerError).send(e).end();
+      }
+    }
+  }
+
+  public async getLoginV2Swif(req: Request, res: Response) {
```
Comment (mattheworris) [inline backend/controllers/AuthController.ts:56]: typo:
```suggestion
  public async getLoginV2Siwf(req: Request, res: Response) {
```

File: backend/services/AccountService.ts [hunk 4]
```diff
@@ -116,9 +138,6 @@ export class AccountService {
           // END REMOVE:
           logger.debug(`Found account for referenceId:(${referenceId})`);
           return {
-            accessToken: createAuthToken(siwfData.accountId),
```
Comment (mattheworris) [inline backend/services/AccountService.ts:119]: Don't we still need an accessToken? Did this move somewhere else?

Comment (mattheworris) [reply backend/services/AccountService.ts:119]: I think this breaks the original sign in with frequency. I'm getting 401 errors after logging in now.

File: backend/services/AccountService.ts [hunk 5]
```diff
@@ -116,9 +138,6 @@ export class AccountService {
...
-            expires: Date.now() + 24 * 60 * 60 * 1_000,
-            referenceId: referenceId,
             msaId: siwfData.msaId,
             handle: response.data.handle as AuthAccountResponse['handle'],
```
Comment (mattheworris) [inline backend/services/AccountService.ts:142]:
```suggestion
            accessToken: createAuthToken(siwfData.accountId),
            expires: Date.now() + 24 * 60 * 60 * 1_000,
            referenceId: referenceId,
            msaId: siwfData.msaId,
            handle: response.data.handle as AuthAccountResponse['handle'],
```

File: backend/services/AccountService.ts (hunk 4 above)
Comment (mattheworris) [reply backend/services/AccountService.ts:119]: I can commit a fix to revert the accessToken changes, let me know if there were other reasons to remove it.

File: frontend/src/features/Login/LoginForm.tsx [hunk 6]
```diff
@@ -46,6 +46,16 @@ interface LoginProps {
 const LoginForm = ({ onLogin, providerId, nodeUrl, siwfUrl }: LoginProps): ReactElement => {
   const [isLoading, setIsLoading] = useState(false);
 
+  const handleLoginSiwaV2 = async () => {
```
Comment (mattheworris) [inline frontend/src/features/Login/LoginForm.tsx:49]: typo?
```suggestion
  const handleLoginSiwfV2 = async () => {
```

**Review by mattheworris — DISMISSED** [review]
- Read through changes
- Tested branch on testnet with ipfs pinning, siwfv1 & siwfv2
- Need to revert accessToken for siwfv1
- A few typos

Nice work!

**Review by mattheworris — APPROVED** [review]

**Pull request merged**